app/Repositories/Eloquent/EmployeeRepository.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace App\Repositories\Eloquent;
+
+use App\Repositories\RepositoryInterface;
+use App\Models\Employee;
+use App\Library\MyValidation;
+use App\Http\Resources\EmployeeCollection;
+use App\User;
+use Validator;
+use App\Library\MyFunctions;
+
+class EmployeeRepository implements RepositoryInterface
+{
+    // model property on class instances
+    protected $model;
+
+    // Constructor to bind model to repo
+    public function __construct()
+    {   
+        $this->model = new Employee();
+    }
+
+    // Get the associated model
+    public function getModel()
+    {
+        return $this->model;
+    }
+
+    // Set the associated model
+    public function setModel($model)
+    {
+        $this->model = $model;
+        return $this;
+    }
+
+    // Get all instances of model
+    public function all()
+    {
+        return new EmployeeCollection($this->model->with('address')->orderBy('id', 'asc')->get());
+    }
+
+    // Get all instances of model with pagination
+    public function paginate($perPage = 10)
+    {
+        return new EmployeeCollection($this->model->with('address')->orderBy('id', 'asc')->paginate($perPage));
+    }
+
+    // create a new record in the database
+    public function create(array $data)
+    {
+        if (User::where('email', $data['email'])->first()) {
```

Comment:
you should use `unique` on validation request. https://laravel.com/docs/5.8/validation#rule-unique